### DISCUSSION TITLE
Welcome to WordPress API for Dart (Dart WP) Discussions!

### DISCUSSION THREAD

**Discussion by theeomm** [dhmgroup/dart-wp#17] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by DenizSAHIN570** [comment]
Could you please post an example for implementation of arguments for retrieving posts?

Edit: I figured out, amazing package saved me from a lot of headache.

**Reply by theeomm** [reply]
Great. I'm glad it helped you out.

**Comment by MohamedMoaayed** [comment]
retrieving media(using .media) returns empty list!?
can you fix that?

**Reply by theeomm** [reply]
Are you still experiencing the same issue in the latest release?

**Reply by MohamedMoaayed** [reply]
The issue is gone with the latest Update.
Thanks.

**Comment by awatson32** [comment]
I'm still pretty new to Flutter, and I can't figure out how to implement this. Can anyone help? I can somehow retrieve what I need but I can't get it to work in a List or anything presentable.

**Reply by DenizSAHIN570** [reply]
Assuming that you know how WordPress API works (incase if you don't know reefer [here](https://developer.wordpress.org/rest-api/)).  

    void fetchData(searchTerm) async {
    
        /// TITLES RETURNED FROM API
        List<String> titles = [];

        /// URLS RETURNED FROM API
        List<String> urls = [];
    
        String searchTerm = term;
        const String searchParam = "_field[]";
        final api = WordPressAPI('https://www.yourwordpresswebsite'); ///WORDPRESS WEBSITE URL
    
        WPResponse res = await api.posts.fetch(args: { ///YOUR ARGUMENTS FOR SPECIFIC SEARCHES
            "search": searchTerm,   
            searchParam: "title",
            searchParam: "url",
            "per_page": "12",
            "orderby": "relevance",
            "order": "asc"
        });

          for (final post in res.data) {
            titles.add(post.title);
            urls.add(post.link);
          }
    }

That is how I ımplemented to retrieve titles based on the search that happened on UI (searchTerm is supplied to the function as a term). After retrieving the data (which is stored in res variable) I get every title and length to an array to list them.

But this is pretty much it is can be used as a simple snippet (there should be try catch for every error response from server like 404 so you can handle errors and show appropriate error message on UI).

**Reply by awatson32** [reply]
Can you show me how to display this? I want to do something like this, but it's saying length isn't defined. 

`  @override
  Widget build(BuildContext context) {
    return FutureBuilder(
      future: fetchPosts(),
      builder: (context, snapshot) {
        return ListView.builder(
            itemCount: snapshot.data!.length,
            itemBuilder: (context, index) {
          return const Text(snapshot.data[index].title);
        });
      },
    );
  }
`

**Reply by DenizSAHIN570** [reply]
I highly suggest that using a state management library such as [Provider](https://pub.dev/packages/provider), so you don't have to deal with returning values (state management libraries do so much more than that and I can not recommend them enough.).

You simply store them as global variables and use their instances, this approach may not be the best practice so take this with a pinch of salt.

I really can't tell how Provider works in single command but for your problem, snapshot may not have arrays since I don't really know what your fetchPosts() returns, or maybe using another way to build the list like below would be more successful.

     List.generate(provide.titles.length,
                              (index) => provide.titles[index].toString())
                          .map((e) => ListTile(
                                onTap: () {
                                },
                                title: Text(
                                  e,
                                  style: const TextStyle(
                                    fontSize: 18,
                                  ),
                                ),
                              ))
                          .toList()

**Comment by Willi180** [comment]
How can I get url from json below. I use argument const arg = {"per_page": "100", "_embed": "1"}; but I cannot get url. Please help. Thanks

`"_embedded": { "author": [ { "id": 2, "name": "test", "url": "http://egmrm.ru", "description": "", ...`

for Example this site https://mensmagazineapp.com/wp-json/wp/v2/posts?_embed
`print(artikelListeFormEgmrm[0].embedModel?.author?.isEmpty);` //artikelListeFormEgmrm[0] is first post

`flutter: true`

but it is not empty

**Comment by bogart0324** [comment]
hi, how can  I get the meta data of custom post type? I tried the fetch but the meta is empty.

**Reply by bogart0324** [reply]
oh ignore this, I manage to figure it out by registering  rest fields. thank you
